src/testdir/test_window_cmd.vim

```diff
@@ -1007,4 +1007,112 @@ func Test_split_cmd()
   close
 endfunc
 
+" Create maximum number of horizontally or vertically split windows and then
+" run commands that create a new horizontally/vertically split window
+func Run_noroom_for_newwindow_test(dir_arg)
+  let dir = (a:dir_arg == 'v') ? 'vert ' : ''
+
+  " Open as many windows as possible
+  for i in range(500)
```

Comment:
500 iterations may not be enough in theory for very large terminals.
I have not tried, but using `while 1 … endwhile` instead would work since the loop should always break with E36.